lib/plotting_utils.py

```diff
@@ -0,0 +1,830 @@
+    if hasattr(data, "units"):
+        ax.set_ylabel("{units}".format(units=getattr(data,"units")))
+    elif "units" in kwargs:
+        ax.set_ylabel("{units}".format(kwargs["units"]))
+    return ax
+
+def zonal_plot_line(ax, lat, data, color, **kwargs):
+    """Create line plot with latitude as the X-axis."""
+    ax = _plot_line(ax, lat, data, color, **kwargs)
+    ax.set_xlim([max([lat.min(), -90.]), min([lat.max(), 90.])])
+    #
+    # annotate
+    #
+    ax.set_xlabel("LATITUDE")
+    if hasattr(data, "units"):
+        ax.set_ylabel("{units}".format(units=getattr(data,"units")))
+    elif "units" in kwargs:
+        ax.set_ylabel("{units}".format(kwargs["units"]))
+    return ax
+
+def zonal_plot_preslat(ax, lat, lev, data, **kwargs):
+    """Create plot with latitude as the X-axis, and pressure as the Y-axis."""
+    mlev, mlat = np.meshgrid(lev, lat)
+    if 'cmap' in kwargs:
+        cmap = kwargs.pop('cmap')
+    else:
+        cmap = 'Spectral_r'
+
+    img = ax.contourf(mlat, mlev, data.transpose('lat', 'lev'), cmap=cmap, **kwargs)
+
+    minor_locator = mpl.ticker.FixedLocator(lev)
+    ax.yaxis.set_minor_locator(minor_locator)
+    ax.tick_params(which='minor', length=4, color='r')
+    ax.set_ylim([np.max(lev), np.min(lev)])
+    return img, ax
+
+
+def meridional_plot_preslon(ax, lon, lev, data, **kwargs):
+    """Create plot with longitude as the X-axis, and pressure as the Y-axis."""
+
+    mlev, mlon = np.meshgrid(lev, lon)
+    if 'cmap' in kwargs:
+        cmap = kwargs.pop('cmap')
+    else:
+        cmap = 'Spectral_r'
+
+    img = ax.contourf(mlon, mlev, data.transpose('lon', 'lev'), cmap=cmap, **kwargs)
+
+    minor_locator = mpl.ticker.FixedLocator(lev)
+    ax.yaxis.set_minor_locator(minor_locator)
+    ax.tick_params(which='minor', length=4, color='r')
+    ax.set_ylim([np.max(lev), np.min(lev)])
+    return img, ax
+
+
+# Color Map Functions
+#--------------------
+
+ncl_defaults = ["ncl_default"]
+
+def guess_ncl_url(cmap):
+    return f"https://www.ncl.ucar.edu/Document/Graphics/ColorTables/Files/{cmap}.rgb"
+
+
+def download_ncl_colormap(url, dest):
+    urlretrieve(url, dest)
+
+
+def read_ncl_colormap(adfobj, fil):
+    # determine if fil is a URL:
+    # if so, we have to download it
+
+    msg = f"\n\t{script_name}: read_ncl_colormap()"
+    if isinstance(fil, str):
+        pars = urlparse(fil)
+        if pars.scheme in ['http', 'https', 'ftp']:
+            filename = Path.cwd() / fil.split("/")[-1]
+            if filename.is_file():
+                msg += f"\n\t\tFile already downloaded as {filename}"
+            else:
+                msg += f"\n\t\tFile will be downloaded and saved as {filename}"
+                download_ncl_colormap(fil, str(filename))
+        else:
+            is_url = False
+            filename = Path(fil)
+    elif isinstance(fil, Path):
+        filename = fil
+    else:
+        raise ValueError(f"\t\tERROR: what to do with type {type(fil)}")
```

Comment:
Instead of killing the ADF run here, would it be better to just send this error message to the debug log and then return `None` like what is done below?  That would in general fit better with the ADF design philosophy of trying to keep the ADF run going even when errors occur.